content/blog/new-graphic-features/index.Rmd

```diff
@@ -0,0 +1,301 @@
+
+```{r}
+grid.circle(
+  gp = gpar(
+    fill = linearGradient(
+      c("firebrick", "steelblue"),
+      x1 = 0, y1 = 0, x2 = 0.5, y2 = 0,
+      extend = "repeat"
+    ),
+    col = NA
+  )
+)
+```
+
+```{r}
+grid.circle(
+  gp = gpar(
+    fill = linearGradient(
+      c("firebrick", "steelblue"),
+      x1 = 0, y1 = 0, x2 = 0.5, y2 = 0,
+      extend = "pad"
+    ),
+    col = NA
+  )
+)
+```
+
+One thing to note is that the extend of the gradient is given relative to the bounding box of the grob being drawn. We could move the circle above around and the gradient would follow along with it.
+
+### Patterns
+
+Like gradients, patterns are a new type of fill made possible in grid through the new features in the graphic engine. Patterns are crazy powerful in that they can be *any* grob you can imagine. The grob itself can consist of other grobs and these grobs could have patterned fill as well (or gradient fills for that matter).
+
+```{r}
+gradient_rec <- rectGrob(
+  width = 0.1, 
+  height = 0.1, 
+  gp = gpar(fill = linearGradient(c("firebrick", "steelblue")))
+)
+grid.draw(gradient_rec)
+```
+
+```{r}
+grid.circle(
+  gp = gpar(fill = pattern(gradient_rec, width = 0.15, height = 0.15, extend = "reflect"))
+)
+```
+
+Understanding the sizing of the grob used for the pattern can take some getting used to. Basically the pattern is drawn relative to the grob. The `width` and `height` argument in the `pattern()` call is then used to define a region of the grob that will be used as a pattern. You can thus not scale the pattern grob using the `width` and `height` argument in `pattern()`. This can be seen below
+
+```{r}
+grid.circle(
+  gp = gpar(fill = pattern(gradient_rec, width = 0.35, height = 0.35, extend = "reflect"))
+)
+```
+
+As we see we are just defining a larger (empty) region from our rect grob, effectively adding more space between each rectangle, rather than creating larger rectangles. This also means that the pattern scales with the grob:
+
+```{r}
+pat <- pattern(
+  gradient_rec, 
+  width = 0.15, 
+  height = 0.15, 
+  extend = "reflect"
+)
+grid.circle(
+  x = 0.25, r = 0.25,
+  gp = gpar(fill = pat)
+)
+grid.circle(
+  x = 0.75, r = 0.5,
+  gp = gpar(fill = pat)
+)
+```
+
+In order to ensure the same scaling of pattern is used across separate grobs, be sure to use absolute units when defining the pattern grob as well as the region:
+
+```{r}
+gradient_rec <- rectGrob(
+  width = unit(1, "cm"), 
+  height = unit(1, "cm"), 
+  gp = gpar(fill = linearGradient(c("firebrick", "steelblue")))
+)
+pat <- pattern(
+  gradient_rec, 
+  width = unit(1.5, "cm"), 
+  height = unit(1.5, "cm"), 
+  extend = "reflect"
+)
+grid.circle(
+  x = 0.25, r = 0.25,
+  gp = gpar(fill = pat)
+)
+grid.circle(
+  x = 0.75, r = 0.5,
+  gp = gpar(fill = pat)
+)
+```
+
+As you can see, patterns can take some getting used to, but this is mainly because the API covers such a span of functionality in terms of sizing etc.
```

Comment:
```suggestion
As you can see, patterns can take some getting used to, but this is mainly because the API covers such a large span of functionality in terms of sizing, etc.
```